shard.go

```diff
@@ -234,7 +234,9 @@ func (s *Shard) close() error {
 	if s.store != nil {
 		_ = s.store.Close()
 	}
-	s.stats.Inc("close")
```

Comment:
I'm not sure how stats can ever be nil.  We instantiate it for both creating or opening a shard.  It feels like somewhere else in our code we are creating a shard and not instantiating it properly?